### DISCUSSION TITLE
Hide credential information included in a task from log files

### DISCUSSION THREAD

**Discussion by z-8-d** [nornir-automation/nornir#745] [Q&A] [answered]
Hi guys! Noob here.
I'm using `nornir` with plugin `nornir_napalm`. I'm using following codes similar to [`nornir_napalm` tests](https://github.com/nornir-automation/nornir_napalm/blob/master/tests/unit/test_napalm_get.py#L81) but with credentials:
```
nr = InitNornir(runner=runner, inventory=inventory, logging=nornir_logging)
nr.run(task=_connect, username=username, password=password, extras=options)
```
Then I found that the plain text of username and password show in log file.
Is there a way to hide these credential information so that they won't show up?

Really appreciate if anyone could help me. Thanks!

**Comment by ktbyers** [comment, marked as answer]
Other people can chime in, but it would be more typical to:

1. Set the username/password in inventory and don't pass them in as arguments at all.
2. Set them directly inside the custom task and once again don't pass them in as arguments.

**Reply by z-8-d** [reply]
> don't pass them in as arguments at all.

This really works! Thanks so much!
